File: app/src/main/java/tv/spring/doc/epub/model/Book.java
package tv.spring.doc.epub.model;

import lombok.Data;
import lombok.extern.log4j.Log4j2;
import org.jetbrains.annotations.NotNull;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.nodes.Node;

@Data
@Log4j2
public class Book {
    private Document document;
    private Element body;

    public Book() {
        document = new Document("");
        body = document.body();
    }

    public void addSection(@NotNull Node node) {
        body.appendChild(node);
    }

}

File: model/src/main/kotlin/tv/spring/doc/epub/model/Book.kt
package tv.spring.doc.epub.model

import org.jsoup.nodes.Document
import org.jsoup.nodes.Element
import org.jsoup.nodes.Node

/**
 * Represents a book with sections.
 *
 * This class provides methods to add sections to a book and retrieve the underlying document.
 */
class Book {
    private val document: Document = Document("")
    private val body: Element = document.body()

    /**
     * Adds a section to the body of the document.
     *
     * @param node The HTML node to be appended to the body.
     */
    fun addSection(node: Node) {
        body.appendChild(node)
    }

    /**
     * Returns the HTML document.
     *
     * @return The HTML document object.
     */
    fun getDocument(): Document {
        return document
    }
}